packages/x-components/src/__tests__/utils.ts

```diff
@@ -125,34 +112,20 @@ export function getMockedAdapter(
   responseFeatures?: Partial<Omit<FeaturesResponseTypes, 'track'>>
 ): MockedSearchAdapter {
   return {
+    /* eslint-disable @typescript-eslint/no-non-null-asserted-optional-chain */
```

Comment:
These types where lying, methods where not returning a `XXXResponse` object if the `responseFeatures.xxx` was undefined. But as this is just testing code, I think we can simplify it and just use non-null assertions rather than over-engineer types and mess up the code. But ask me once and I'll do it.

Reply:
I wouldn't waste time with this, as we're going to refactor the adapter anytime soon(_tm_)